Discovery/Discovery/Presentation/DiscoveryView.swift

```diff
@@ -145,12 +154,19 @@ public struct DiscoveryView: View {
                 }
             }
         }
+        .navigationBarHidden(fromStartupScreen ? false : true)
         .onFirstAppear {
+            if !(viewModel.searchQuery?.isEmpty ?? true) {
+                router.showDiscoverySearch(searchQuery: viewModel.searchQuery)
+            }
             Task {
                 await viewModel.discovery(page: 1)
             }
             viewModel.setupNotifications()
         }
+        .onDisappear {
+            viewModel.searchQuery = ""
+        }
```

Comment:
Maybe we can set "" to searchQuery just after calling the router.showDiscoverySearch function?

Reply:
Done